### PR TITLE
Admin method changes 

### PR DESCRIPTION
**Pull request by rahulKQL**
### Commit Contains: 
Veneer client related changes for `AbstractBigtableAdmin` which is tracked under #1974 

Adding updates for following methods:
- [x] createTableAsync
- [x] addColumn
- [x]  modifyColumn
- [x]  deleteColumn
- [x] createTable
- [x]  modifyColumns
- [x] deleteColumn

Also Added `ColumnDescriptorAdapter#buildGarbageCollectionRule` for `com.google.cloud.bigtable.admin.v2.models.GCRules.GCRule` creation.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/admin/ColumnDescriptorAdapter.java [hunk 1]
```diff
@@ -358,4 +320,46 @@ public HColumnDescriptor adapt(String familyName, ColumnFamily columnFamily) {
...
     return hColumnDescriptor;
   }
+
+  /**
+   * Construct an Bigtable {@link com.google.cloud.bigtable.admin.v2.models.GCRules.GCRule} from the
+   * given column descriptor.
+   * @param columnDescriptor a {@link org.apache.hadoop.hbase.HColumnDescriptor} object.
+   * @return a {@link com.google.cloud.bigtable.admin.v2.models.GCRules.GCRule} object.
+   */
+  public static GCRule buildGarbageCollectionRule(HColumnDescriptor columnDescriptor) {
```
[the remark this answers is not in the document]
Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/admin/ColumnDescriptorAdapter.java]: sure, Will do it now.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java [hunk 2]
```diff
@@ -159,11 +180,18 @@ public void createTable(TableDescriptor desc, byte[][] splitKeys) throws IOExcep
    * Calling {@link #addColumn(TableName, ColumnFamilyDescriptor)} was causing stackoverflow.
    * Copying the same code here. //TODO - need to find a better way
    */
+  // TODO(rahulkql): confirm if above case is still valid
   @Override
   public void addColumnFamily(TableName tableName, ColumnFamilyDescriptor columnFamilyDesc)
       throws IOException {
-    modifyColumns(tableName, columnFamilyDesc.getNameAsString(), "add",
-        ModifyTableBuilder.create().add(TableAdapter2x.toHColumnDescriptor(columnFamilyDesc)));
+    com.google.cloud.bigtable.admin.v2.models.ModifyColumnFamiliesRequest modifyColumnRequest =
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java]: Please use `com.google.cloud.bigtable.admin.v2.models.ModifyColumnFamiliesRequest` as the import, and change all of the proto `ModifyColumnFamiliesRequest` uses to use the model as well.

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 3]
```diff
@@ -348,13 +355,31 @@ public void createTable(HTableDescriptor desc, byte[] startKey, byte[] endKey, i
   /** {@inheritDoc} */
   @Override
   public void createTable(HTableDescriptor desc, byte[][] splitKeys) throws IOException {
-    createTable(desc.getTableName(), TableAdapter.adapt(desc, splitKeys));
+    CreateTableRequest createReq = CreateTableRequest.of(desc.getTableName().getNameAsString());
+
+    for (HColumnDescriptor column : desc.getColumnFamilies()) {
+      String columnName = column.getNameAsString();
+      // GcRule gcRuleProto = buildGarbageCollectionRule(column);
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Please remove the comment.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java (hunk 2 above)
Comment (rahulKQL) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java]: Updated imports for `models. ModifyColumnFamiliesRequest`

**Comment by rahulKQL** [comment]
@sduskis Could you please have a look it.

**Comment by rahulKQL** [comment]
@sduskis Whenever you get a chance, please have a look at it.

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 4]
```diff
@@ -348,13 +355,29 @@ public void createTable(HTableDescriptor desc, byte[] startKey, byte[] endKey, i
   /** {@inheritDoc} */
   @Override
   public void createTable(HTableDescriptor desc, byte[][] splitKeys) throws IOException {
-    createTable(desc.getTableName(), TableAdapter.adapt(desc, splitKeys));
+    CreateTableRequest createReq = CreateTableRequest.of(desc.getTableName().getNameAsString());
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Why isn't this done in `TableAdapter`?

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 5]
```diff
@@ -364,14 +387,30 @@ protected void createTable(TableName tableName, CreateTableRequest.Builder build
   @Override
   public void createTableAsync(final HTableDescriptor desc, byte[][] splitKeys) throws IOException {
     LOG.warn("Creating the table synchronously");
-    CreateTableRequest.Builder builder = TableAdapter.adapt(desc, splitKeys);
-    createTableAsync(builder, desc.getTableName());
+    CreateTableRequest createTableRequest =
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Same here.  There's duplication.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java [hunk 6]
```diff
@@ -104,7 +108,21 @@ public void createTable(TableDescriptor desc, byte[] startKey, byte[] endKey, in
   /** {@inheritDoc} */
   @Override
   public void createTable(TableDescriptor desc, byte[][] splitKeys) throws IOException {
-    createTable(desc.getTableName(), TableAdapter2x.adapt(desc, splitKeys));
+    CreateTableRequest createTableRequest =
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java]: Same here.

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java (hunk 4 above)
Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Thanks for the feedback. Previously I thought of migrating  `TableAdapter` with `v2.models.CreateTableRequest` later, now that is also updated.

**Comment by sduskis** [comment]
Can you please create a separate PR for the `TableAdapter`?  There's too much going on in this PR for a good review.

**Comment by rahulKQL** [comment]
sure, I will create separate PR for both `TableAdapter`  &  `TableAdapter2x`

**Comment by rahulKQL** [comment]
Closing this PR, Reason as rebase went wrong.  
I will create another PR with specific changes only.

**Pull request closed without merging**